app/assets/javascripts/host_checkbox.js

```diff
@@ -63,7 +63,7 @@ $(function() {
 });
 
 function removeForemanHostsCookie() {
-  $.cookie($.cookieName, null);
+  $.removeCookie($.cookieName, { path: '/' });
```

Comment:
it is that the cookie is valid on (https://github.com/carhartl/jquery-cookie#path).
``` path: '/' ``` means the entire domain.